analyzers/rolaguard_base_analyzer/CheckPacketsLost.py

```diff
@@ -0,0 +1,71 @@
+from collections import defaultdict
+import logging as log
+
+from utils.AlertGenerator import emit_alert
+from db.Models import  DeviceCounters, CounterType
+
+
+
+class CheckPacketsLost():
+    def __init__(self):
+        # Dict structure that stores device ids as keys and estimated number of lost packets as values.
+        # Also, it stores the date in which the last update for this value has been made, to be used
+        # in garbage collection
+        # Works like a cached version of lost packets counter stored in database (device_counters table)
+        self.devices_packets_lost = defaultdict(lambda: {})
+        self.last_gc = None
+
+    def __call__(self, packet, device_session, device, gateway, policy_manager):
+        if not self.last_gc: self.last_gc = packet.date
+        if (packet.date - self.last_gc).seconds > 3600: # Garbage collect every hour
+            self.garbage_collection(today = packet.date) 
+
+        if packet.npackets_lost_found > 0:
+            # Initialize dict entry if it not exists, increment it otherwise
+            if device.id not in self.devices_packets_lost:
+                stored_packets_lost = DeviceCounters.get_device_counter(
+                    device_id=device.id,
+                    packet_date=packet.date,
+                    counter_type=CounterType.PACKETS_LOST,
+                    window=policy_manager.get_parameters("LAF-101")["time_window"]
+                )
+
+                self.devices_packets_lost[device.id]["packets_lost"] = stored_packets_lost
+            else:
+                self.devices_packets_lost[device.id]["packets_lost"] += 1
```

Comment:
Here packets_lost must be incremented by packet.npackets_lost_found, not by one.

Reply:
You're right